Jenkinsfile

```diff
@@ -82,79 +83,173 @@ spec:
-                }
-            }
-            stage ('Create project') {
-                if ( currentBuild.result != null ) { stages_failed = true; return; }
-                catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
-                    openshift.withCluster(){
-                        openshift.newProject(namespace)
-                    }
-                }
-            }
-            stage('Build STF Containers') {
-                if ( currentBuild.result != null ) { stages_failed = true; return; }
-                catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
-                    ansiColor('xterm') {
-                        ansiblePlaybook(
-                            // use the playbook to build the containers but don't run CI
-                            playbook: 'build/run-ci.yaml',
-                            colorized: true,
-                            extraVars: [
-                                "namespace": namespace,
-                                "__deploy_stf": "false",
-                                "__local_build_enabled": "true",
-                                "__service_telemetry_snmptraps_enabled": "true",
-                                "__service_telemetry_storage_ephemeral_enabled": "true",
-                                "working_branch":"${working_branch}"
-                            ]
-                        )
-                    }
-                }
-            }
-            stage('Deploy STF Object') {
-                if ( currentBuild.result != null ) { stages_failed = true; return; }
-                catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
-                    openshift.withCluster() {
-                        openshift.withProject(namespace) {
-                            timeout(time: 800, unit: 'SECONDS') {
-                                openshift.create(stf_resource)
-                                sh "OCP_PROJECT=${namespace} ./build/validate_deployment.sh"
-                            }
-                        }
-                    }
-                }
-            }
-            stage('Run Smoketest') {
-                if ( currentBuild.result != null ) { stages_failed = true; return; }
-                catchError(buildResult: 'FAILURE', stageResult: 'FAILURE') {
-                    sh "OCP_PROJECT=${namespace} ./tests/smoketest/smoketest.sh"
-                }
-            }
-            stage('Cleanup') {
-                openshift.withCluster(){
-                    openshift.selector("project/${namespace}").delete()
-                    if ( stages_failed ) { currentBuild.result = 'FAILURE' }
-                }
-                if ( stages_failed ) { currentBuild.result = 'FAILURE' }
-            }
-        }
-    }
+pipeline {
+	agent {
+		kubernetes {
+			inheritFrom 'ocp-agent'
+			defaultContainer 'exec'
+		}
+	}
+	stages {
+		stage('Clone Upstream') {
+			when {
```

Comment:
Is there any way to avoid re-writing this condition in every stage?

Reply:
Unfortunately the "when" block must be inside a "stage" block, we can't move it outside and apply it for all stages at once. Possible solution might be to enclose all of our current stages into one top level stage and put the condition there. Something similar to the example there: https://www.jenkins.io/doc/book/pipeline/syntax/#sequential-stages

We would basically go from:
```
stages {
    stage('1') {
        when {
            condition
        }
        steps {
            ...
        }
    }
    stage('1') {
        when {
            condition
        }
        steps {
            ...
        }
    }
}
```
to:
```
stages {
    stage('top-stage') {
        when {
            condition
        }
        stages {
            stage('1') {
                steps {
                    ...
                }
            }
            stage('2') {
                steps {
                    ...
                }
            }
        }
    }
}
```
But I think the currentBuild.result == null might still be needed at the beginning of each stage.